src/ralph/attachments/forms.py

```diff
@@ -27,6 +30,31 @@ def save(self, commit=True):
             * mime_type - uploaded file's content type.
         """
         obj = super().save(commit=False)
+        md5 = Attachment.get_md5_sum(obj.file)
+        attachment = Attachment.objects.filter(md5=md5)
+        if obj.pk:
+            attachment = attachment.exclude(pk=obj.pk)
+        attachment = attachment.first()
+        if attachment:
+            content_type = ContentType.objects.get_for_model(
+                self._parent_object._meta.model
+            )
+            if not AttachmentItem.objects.filter(
+                content_type=content_type,
+                object_id=self._parent_object.pk,
+                attachment__md5=md5
+            ).exists():
+                AttachmentItem.objects.attach(
+                    self._parent_object.pk, content_type, [attachment]
+                )
+                return None
+            message = _('File %(filename)s is attached already.') % {
```

Comment:
do we really need to inform user about our internal details (that we don't store duplicated attachments)?